lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -297,3 +244,131 @@ def _get_org_key_names(self):
+            # This is initial rendering state.
+            pass
+        else:
+            search_error = (
+                "To perform a search, you must provide either the student's "
+                "(a) edX username, "
+                "(b) email address associated with their edX account, or "
+                "(c) Identity-providing institution and external key!"
+            )
+
+        return render_to_response(
+            self.CONSOLE_TEMPLATE_PATH,
+            {
+                'error': search_error,
+                'learner_program_enrollments': learner_program_enrollments,
+                'org_keys': sorted(saml_providers_with_org_key.keys()),
+            }
+        )
+
+
+class ProgramEnrollmentsInspectorAPIView(ProgramEnrollmentInspector, APIView):
+    """
+    The APIview to search and display the program enrollments
+    information of a learner.
+    """
+
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        """
+        Based on the query string parameters passed through the GET request
+        Search the data store for information about ProgramEnrollment and
+        SSO linkage with the user.
+        * Example Request:
+            - GET / support / program_enrollments_inspector_details /
+            * Sample Payload
+            {
+                edx_user: < edx_user > ,
+                org_key: < org_key >
+                external_user_key: < external_user_key >
+            }
+        * Example Response:
+            {
+                learner_program_enrollments: < learner_program_enrollments >,
+                org_key: < org_key >
+                errors: 'Error messages for invalid query'
+            }
+        """
+        search_error = ''
+        edx_username_or_email = request.GET.get('edx_user', '').strip()
+        org_key = request.GET.get('org_key', '').strip()
+        external_user_key = request.GET.get('external_user_key', '').strip()
+        learner_program_enrollments = {}
+        saml_providers_with_org_key = self._get_org_keys_and_idps()
+        selected_provider = None
+        if org_key:
+            selected_provider = saml_providers_with_org_key.get(org_key)
+        if edx_username_or_email:
+            learner_program_enrollments, search_error = self._get_account_info(
+                edx_username_or_email,
+                selected_provider,
+            )
+        elif org_key and external_user_key:
+            learner_program_enrollments = self._get_external_user_info(
+                external_user_key,
+                org_key,
+                selected_provider,
+            )
+            if not learner_program_enrollments:
+                search_error = 'No user found for external key {} for institution {}'.format(
+                    external_user_key, org_key
+                )
+        elif not org_key and not external_user_key:
+            # This is initial rendering state.
+            pass
```

Comment:
do we need this here, considering we are returning an API response?

Reply:
No, it should throw search errors. I was under impression that it shouldn't throw such an error when nothing is provided.